pkg/gofr/migration/mongo.go

```diff
@@ -0,0 +1,100 @@
+package migration
+
+import (
+	"context"
+	"time"
+
+	"gofr.dev/pkg/gofr/container"
+)
+
+type mongoDS struct {
+	container.Mongo
+}
+
+type mongoMigrator struct {
+	container.Mongo
+	migrator
+}
+
+// apply initializes mongoMigrator using the Mongo interface.
+func (ds mongoDS) apply(m migrator) migrator {
+	return mongoMigrator{
+		Mongo:    ds.Mongo,
+		migrator: m,
+	}
+}
+
+const (
+	mongoMigrationCollection = "gofr_migrations"
+)
+
+// checkAndCreateMigrationTable initializes a MongoDB collection if it doesn't exist.
+func (mg mongoMigrator) checkAndCreateMigrationTable(_ *container.Container) error {
+	err := mg.Mongo.CreateCollection(context.Background(), mongoMigrationCollection)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getLastMigration retrieves the latest migration version from MongoDB.
+func (mg mongoMigrator) getLastMigration(c *container.Container) int64 {
+	var lastMigration int64
+
+	var migrations []struct {
+		Version int64 `bson:"version"`
+	}
+
+	filter := map[string]interface{}{}
+
+	err := mg.Mongo.Find(context.Background(), mongoMigrationCollection, filter, &migrations)
+	if err != nil {
+		c.Errorf("Failed to fetch migrations from MongoDB: %v", err)
+		return 0
+	}
+
+	// Identify the highest migration version.
+	for _, migration := range migrations {
+		if migration.Version > lastMigration {
+			lastMigration = migration.Version
+		}
```

Comment:
max is now a built-in function

```suggestion
		lastMigration = max(lastMigration, migration.Version)
```